### DISCUSSION TITLE
Is autocomplete for HTML elements available?

### DISCUSSION THREAD

**Discussion by francisceril** [zed-industries/zed#6548] [Feature Requests]
Working on some .html and .tsx files and neither have auto completion for any HTML element. Is this feature not available yet?

**Comment by hovsater** [comment]
Auto completion is provided by the respective language servers. It seems to be working fine for me. See the screenshot below.

<img width="449" alt="Screenshot 2023-04-27 at 21 41 44" src="https://user-images.githubusercontent.com/503025/234973781-a970e921-0828-463e-b277-66031613baff.png">

If you press `⌘⇧P` and enter `zed: open log` does it tell you anything interesting? Perhaps you're getting an error from the language server. We've had cases were the language servers got corrupted during download. In that case, try the following:

1. Close Zed completely
2. In a terminal of your choice run the following commands:
```console
$ rm -rf ~/Library/Application\ Support/Zed/node ~/Library/Application\ Support/Zed/languages/vscode-html-language-server ~/Library/Application\ Support/Zed/languages/typescript-language-server
```
3. Open Zed and try again.

**Reply by francisceril** [reply]
Alright so I just did the `rm -rf ..` and got the suggestions showing but not really the autocomplete I was expecting. Currently I have to type the opening `<` for the suggestions to show. It would be nice if I don't have to type that and would automatically close the tag.
Right now, these are just suggestions and not autocomplete.

**Reply by hovsater** [reply]
This isn't something Zed currently supports. Feel free to open an issue describing this enhancement. 🙂

**Reply by mpassell** [reply]
@francisceril It looks like auto-closing the tags is already covered by the issue https://github.com/zed-industries/community/issues/392 but I don't see anything for autocomplete, so you could make one just for that.

**Reply by francisceril** [reply]
Emmet is already requested in this issue #993 and that pretty much does what I wanted unless anyone needs autocomplete for other languages.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
